### DISCUSSION TITLE
[20240601] Player buttons behave unexpected, Scrobler not working, playlist unresponsive

### DISCUSSION THREAD

**Discussion by ha11oga11o** [rern/rAudio#1678] [Fixed Issues] [answered]
Hello,

i had some issues with playlist not responsive where i create bug report #1661 . Its still affected, and i observed other issues since 20240601 update.

It seems player buttons are not all working. Skip forward or back is not working at all.
LastFm scrobble is broken also. I did manual push with

/srv/http/bash/scrobble.sh "cmd
Pink Floyd
Time
CMD ARTIST TITLE"

and i got it reflected, but from player itself its not working anymore. 

https://www.last.fm/user/ha11oga11o

Heres video for better clarification: 

https://github.com/rern/rAudio/assets/1357309/45fd953f-9a56-4d51-aae9-c434cc0c4aa6

rAudio 20240601
6.1.58-2 aarch64
Raspberry Pi 3B+ Rev 1.3
BCM2837B0 • 906 MiB
4 x Cortex-A53 @ 1.40 GHz

**Comment by ha11oga11o** [comment]
Also,
I tried other browsers, other networks for web access but no results. Only app i found where all things works fine is with this app. Except scrobbler of course. 

 
![IMG_7828](https://github.com/rern/rAudio/assets/1357309/301c40f6-c7b5-4194-8773-29e315de4126)

Thanks

**Comment by rern** [comment]
It might be an issue with metadata tags of some tracks. I've added 10,000 tracks to Playlist and it ran fine.
Please try adding one album at a time until the issue occurs then verify all tags of the album.

**Comment by ha11oga11o** [comment]
I did some more research on playlist not working issue, and i manage to see in browser that im getting http 500 error. 

I will try to paste at least most important ones here (im sorry but i simply cant hit formatting right, so i paste it raw).

http://192.168.1.216/mpdplaylist.php
Status
500
Internal Server Error
VersionHTTP/1.1
Transferred851 B (0 B size)
Referrer Policystrict-origin-when-cross-origin
Request PriorityHighest
DNS ResolutionSystem

	
Connection
	keep-alive
Content-Type
	text/html; charset=UTF-8
Date
	Tue, 04 Jun 2024 05:51:48 GMT
Server
	nginx/1.25.2
Transfer-Encoding
	chunked
X-Powered-By
	PHP/8.2.11
	
Accept
	application/json, text/javascript, */*; q=0.01
Accept-Encoding
	gzip, deflate
Accept-Language
	en-US,en;q=0.5
Connection
	keep-alive
Content-Length
	16
Content-Type
	application/x-www-form-urlencoded; charset=UTF-8
Cookie
	_ga_EMK40SCD1G=GS1.1.1698649592.2.1.1698650393.0.0.0; _ga=GA1.1.325424401.1698040873; io=OkB5V0Z_x7QQf_dJAAAk
Host
	192.168.1.216
Origin
	http://192.168.1.216
Priority
	u=1
Referer
	http://192.168.1.216/
Sec-GPC
	1
User-Agent
	Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
X-Requested-With
	XMLHttpRequest

After 01.06. update i have same error on skip buttons too (as shown on video) tracks too. I dont dubt that theres file, or two,... or even more with bad metadata and probably thats causing problems. But is there any chance to add some error log, popup, anything just to point out whats an issue? 
At least for rpi3 is bit hard to populate huge lists due slow hardware, and without any return info its kinda impossible to pin point error. 

Thank you. 

edit: since 20240603 it seems scrobler works, but buttons for me are still unresponsive.

**Reply by rern** [reply]
Files with non UTF-8 metadata tags, if any, will appear in:
Settings > Player > Non UTF-8 Files (click info icon)

**Comment by ha11oga11o** [comment]
I hope thats info i need? I see no errors thou. Player plays without errors. IT just not loading playlist in web browser. 

`# /srv/http/data/mpdconf/mpd.conf
bind_to_address         "0.0.0.0"
db_file                 "/srv/http/data/mpd/mpd.db"
max_output_buffer_size  "24576"
music_directory         "/mnt/MPD"
playlist_directory      "/srv/http/data/playlists"
state_file              "/var/lib/mpd/mpdstate"
user                    "mpd"
input {
	plugin  "curl"
}
decoder {
	plugin   "ffmpeg"
	enabled  "yes"
}
audio_output {
	type       "httpd"
	name       "Streaming"
	encoder    "flac"
	port       "8000"
	quality    "5.0"
	format     "44100:16:2"
	always_on  "yes"
}
audio_output {
	name           "On-board Headphones"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "hardware"
	mixer_control  "PCM"
	mixer_device   "hw:0"
}

# systemctl status mpd
* mpd.service - Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: active (running) since Tue 2024-06-04 07:48:21 CEST; 6h ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
   Main PID: 476 (mpd)
      Tasks: 7 (limit: 1006)
        CPU: 13min 38.383s
     CGroup: /system.slice/mpd.service
             └─476 /usr/bin/mpd --systemd /srv/http/data/mpdconf/mpd.conf

Jun 04 12:14:21 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/11 - deutsch nepal - rolf zacher vocals(bonus).mp3"
Jun 04 12:17:20 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/12 - goldrush(bonus).mp3"
Jun 04 12:21:53 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/13 - star eyed(bonus).mp3"
Jun 04 12:25:41 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/14 - dr. stein(bonus).mp3"
Jun 04 12:33:05 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1977 - amon düül ii - almost alive/01 - One Blue Morning.mp3"
Jun 04 12:41:15 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1977 - amon düül ii - almost alive/02 - Good Bye My Love.mp3"
Jun 04 12:48:32 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1977 - amon düül ii - almost alive/03 - Ain't Today Tomorrow's Yesterday.mp3"
Jun 04 12:52:48 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1977 - amon düül ii - almost alive/04 - Hallelujah.mp3"
Jun 04 13:06:02 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1977 - amon düül ii - almost alive/06 - Live in Jericho.mp3"
Jun 04 13:12:10 rAudio mpd[476]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1977 - amon düül ii - almost alive/05 - Feeling Uneasy.mp3"
 `

**Reply by rern** [reply]
Try: `php /srv/http/mpdplaylist.php current`

**Comment by ha11oga11o** [comment]
This is output of this command

rAudio:~  #  php /srv/http/mpdplaylist.php current
PHP Fatal error:  Uncaught TypeError: Unsupported operand types: int + string in /srv/http/mpdplaylist.php:194
Stack trace:
#0 /srv/http/mpdplaylist.php(93): htmlTrack()
#1 /srv/http/mpdplaylist.php(19): currentPlaylist()
#2 {main}
  thrown in /srv/http/mpdplaylist.php on line 194
rAudio:~  #

**Reply by rern** [reply]
`Time` tag is missing or wrong.
Try: `sed -i 's/$counttime +=/if ( is_int( $sec ) ) &/' /srv/http/mpdplaylist.php` 
Then tap Playlist again.

**Comment by ha11oga11o** [comment]
That works fine,

finally whole playlist and its cool features are there. Is this permanently fixed or i need to apply this line every time i create "rouge" playlist? 

Now, even skip to next track buttons works. You are THE best! It was splint in my spine for some time. 

Thank you!

**Reply by rern** [reply]
Please post the output of: `mpc playlist -f %file%^%time% | grep -v ^http | grep -m20 [^0-9]$`

**Comment by ha11oga11o** [comment]
Here is output,

rAudio:~  #  mpc playlist -f %file%^%time% | grep -v ^http | grep -m20 [^0-9]$
NAS/NasData/Media/Music/Strana/Anthology/Livin' Blues/Livin' Blues 2/2012 - Now 25 Year Anniversary/Livin' Blues - Now 25 Year Anniversary.cue/track0006^
rAudio:~  #

**Comment by ha11oga11o** [comment]
All issues here fixed with 20240610.

Thanks

**Reply by rern** [reply, marked as answer]
Something's wrong with your `NAS/NasData/Media/Music/Strana/Anthology/Livin' Blues/Livin' Blues 2/2012 - Now 25 Year Anniversary/Livin' Blues - Now 25 Year Anniversary.cue` > `TRACK 06` > `INDEX`.
